Extension/src/common.ts

```diff
@@ -214,6 +215,22 @@ export async function isExtensionReady(): Promise<boolean> {
     return doesInstallLockFileExist;
 }
 
+export function renameDebugAdaptersPath(): Promise<void> {
+    if (os.platform() !== 'win32') {
+        const sourcePath: string = getDebugAdaptersPath("bin/OpenDebugAD7.exe.config");
+        if (fs.existsSync(sourcePath)) {
+            fs.rename(sourcePath, getDebugAdaptersPath("bin/OpenDebugAD7.exe.config.unused"), (err: NodeJS.ErrnoException | null) => {
+                if (err) {
+                    getOutputChannelLogger().appendLine(localize("rename.failed.delete.manually",
+                        'ERROR: fs.rename failed with "{0}". Delete {1} manually to enable debugging.', err.message, sourcePath));
+                }
+            });
+        }
+    }
+
+    return Promise.resolve();
```

Comment:
It doesn't look like this function returns a Promise other than this call to `Promise.resolve()`.  Maybe this function doesn't need to return a Promise?  There is a helper function that wraps `fs.rename()` in common.ts, called `renamePromise`.  It's async.  I'd suggest calling `await renamePromise()` instead of using the callback, making `renameDebugAdaptersPath` `async`, and omitting the call to `Promose.resolve()` (as an async function implicitly returns a Promise).

Reply:
It probably had a promise because of the callback for rename. If that is the case, then this code needs to be wrapped in a new Promise and you call resolve when the rename callback is invoked.

Reply:
Wouldn't it be better to `await renamePromise()` instead, and make this function async?  I like the idea of avoiding creation of new Promises, pushing them down into async helpers.  (Also, IMHO, I'd rename `renamePromise` to `renameAsync`(?))

Reply:
I made the change. However, I can't see the function `renameDebugAdaptersPath `is being called anywhere.

Reply:
I don't see where `renameDebugAdaptersPath` is being called either.  Isn't it added in this PR?  Can it be removed?

Reply:
@Colengc should I remove it? It was not added in this PR (it was an existing code).

Reply:
I'm confused.  In the diff it shows as green (newly added), which means you are adding this function.  I don't see any red section in the diff indicating it was moved, or any removed files.  I assumed you had written it.  Maybe this is left over from a bad merge?  I do see a function `removeUnnecessaryFile()` in main.ts, which looks to be the same function, and that is called in main.ts in 2 places.  If this one is not used, I'd suggest removing it.

Reply:
I removed it